hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java

```diff
@@ -148,7 +148,18 @@ public void readVolume(File hddsVolumeRootDir) {
       LOG.info("Start to verify containers on volume {}", hddsVolumeRootDir);
       File currentDir = new File(idDir, Storage.STORAGE_DIR_CURRENT);
       File[] containerTopDirs = currentDir.listFiles();
-      if (containerTopDirs != null) {
+      if (containerTopDirs != null && containerTopDirs.length > 0) {
+        try {
+          // idDir is working directory having data
+          // and volume is initialized with temp path
+          hddsVolume.createTmpDirs(idDir.getName());
+        } catch (IOException e) {
+          LOG.error("Volume {} dir path {} can not be accessed to create " +
```

Comment:
Can we refactor something like,
`LOG.error("The tmp directory can't be created inside {} in the " +
                  "volume {}.", idDir.getName(), hddsVolumeRootDir, e);`